### PR TITLE
Addition of Clonedigger Reports, One Clone Removed

### PR DESCRIPTION
**Pull request by JPIvan**
Created a folder containing two clonedigger reports, one before and one after my changes to utilsplot.py.

Changes: rga_plot function duplicated code for plotting with plot_type='input' and plot_type='output' parameter removed

### CODE REVIEW

File: utilsplot.py [hunk 1]
```diff
@@ -573,6 +573,18 @@ def rga_plot(G, w_start=-2, w_end=2, axlim=None, points=1000, fig=0,
     freqresp = [G(si) for si in s]
 
     plot_No = 1
+
```
Comment (alchemyst) [inline utilsplot.py]: Please make sure your syntax conforms to PEP8. The spacing around this dictionary is wonky.

File: utilsplot.py [hunk 2]
```diff
@@ -601,19 +613,22 @@ def rga_plot(G, w_start=-2, w_end=2, axlim=None, points=1000, fig=0,
                 box = ax.get_position()
                 ax.set_position([box.x0, box.y0,
                                  box.width * 0.8, box.height * 0.9])
-
-    elif plot_type == 'outputs':  # i
+    
+    elif plot_type in plotting_order.keys():
```
Comment (alchemyst) [inline utilsplot.py]: `in dictionary` does the same as `in dictionary.keys()`

File: utilsplot.py [hunk 3]
```diff
@@ -601,19 +613,22 @@ def rga_plot(G, w_start=-2, w_end=2, axlim=None, points=1000, fig=0,
...
-            ax = fig.add_subplot(dim[1], 1, plot_No)
-            ax.set_title('Output %s vs. input j' % (i + 1))
+                           'RGA magnitude',
+                           -0.05, -0.03, 1, 0.9)
+        for i in range( plotting_order[plot_type]['i'] ):
+            ax = fig.add_subplot( plotting_order[plot_type]['j'], 1, plot_No)
+            ax.set_title(plotting_order[plot_type]['title'] % (i + 1))
             rgamax = []
-            for j in range(dim[1]):
+            for j in range( plotting_order[plot_type]['j'] ):
```
Comment (alchemyst) [inline utilsplot.py]: PEP8 doesn't want extra spaces between parentheses

**Comment by JPIvan** [comment]
As a general rule we do not put generated files in the repository as this leads to churn. So please don't check in the clone digger reports.

	I haven't worked on anything like this before, so please excuse me if I make further errors of this sort.
	Clone digger reports removed. The logic for adding them was that I had to create a python 2 environment to generate them as clone digger does not currently support python 3.

PEP8 Standards:

	I have made some edits to adhere to the PEP8 style guide.
	
	https://www.python.org/dev/peps/pep-0008/#indentation
	
	my_list = [
		1, 2, 3,
		4, 5, 6,
	]
	result = some_function_that_takes_arguments(
		'a', 'b', 'c',
		'd', 'e', 'f',
	)
	
	Is now the convention followed.

In utilsplot.py:

> @@ -573,6 +573,18 @@ def rga_plot(G, w_start=-2, w_end=2, axlim=None, points=1000, fig=0,
     freqresp = [G(si) for si in s]
 
     plot_No = 1
+    
Please make sure your syntax conforms to PEP8. The spacing around this dictionary is wonky.

	Addressed.

In utilsplot.py:

> @@ -601,19 +613,22 @@ def rga_plot(G, w_start=-2, w_end=2, axlim=None, points=1000, fig=0,
                 box = ax.get_position()
                 ax.set_position([box.x0, box.y0,
                                  box.width * 0.8, box.height * 0.9])
-
-    elif plot_type == 'outputs':  # i
+    
+    elif plot_type in plotting_order.keys():
in dictionary does the same as in dictionary.keys()

	I was unaware, corrected.

In utilsplot.py:

>              rgamax = []
-            for j in range(dim[1]):
+            for j in range( plotting_order[plot_type]['j'] ):
PEP8 doesn't want extra spaces between parentheses

	Addressed.
	
---

	All string formatting performed with % operator replaced with str.format() method as both instances were present. str.format() chosen because of easier implementation of nested parameters in math format strings.
	
	Reduced some line lengths by separating nested function calls.
	
	Blocked code for clarity.

File: utilsplot.py (hunk 1 above)
Comment (JPIvan) [reply utilsplot.py]: **Addressed.**

File: utilsplot.py (hunk 2 above)
Comment (JPIvan) [reply utilsplot.py]: **I was unaware, corrected.**

File: utilsplot.py (hunk 3 above)
Comment (JPIvan) [reply utilsplot.py]: **Addressed.**

File: utilsplot.py [hunk 4]
```diff
@@ -573,6 +573,21 @@ def rga_plot(G, w_start=-2, w_end=2, axlim=None, points=1000, fig=0,
     freqresp = [G(si) for si in s]
 
     plot_No = 1
+    
+    plotting_order = {
+        'outputs' : {
```
Comment (alchemyst) [inline utilsplot.py]: Spacing is still strange. Should be `'value': key` (no space before colon, one space after) everywhere.

**Comment by JPIvan** [comment]
I've had the code checked automatically for PEP8 violations. Violations in rga_plot() reported by pycodestlye:

> 2:1: E302 expected 2 blank lines, found 1
> 32:80: E501 line too long (136 > 79 characters)
> 43:1: W293 blank line contains whitespace
> 45:18: E203 whitespace before ':'
> 46:16: E231 missing whitespace after ':'
> 47:16: E231 missing whitespace after ':'
> 48:20: E231 missing whitespace after ':'
> 49:21: E231 missing whitespace after ':'
> 51:17: E203 whitespace before ':'
> 52:16: E231 missing whitespace after ':'
> 53:16: E231 missing whitespace after ':'
> 54:20: E231 missing whitespace after ':'
> 55:21: E231 missing whitespace after ':'
> 71:1: W293 blank line contains whitespace
> 74:1: W293 blank line contains whitespace
> 76:1: W293 blank line contains whitespace
> 87:1: W293 blank line contains whitespace
> 89:80: E501 line too long (81 > 79 characters)
> 95:80: E501 line too long (80 > 79 characters)
> 99:1: W293 blank line contains whitespace
> 101:1: W293 blank line contains whitespace
> 107:1: W293 blank line contains whitespace
> 109:1: W293 blank line contains whitespace
> 113:1: W293 blank line contains whitespace
> 118:80: E501 line too long (84 > 79 characters)

All addressed.

File: utilsplot.py (hunk 4 above)
Comment (JPIvan) [reply utilsplot.py]: Now PEP8 compliant.

**Comment by alchemyst** [comment]
Thanks for your efforts. I know it can be frustrating to do a good thing only to be told you didn't do it quite right. FYI PyCharm will highlight most of these problems and automatically fix them if you go to "Code|Inspect code".

**Comment by JPIvan** [comment]
No no, this is your codebase, I want to contribute only something that you are completely happy with. I understand the need to be thorough.

**Pull request merged**